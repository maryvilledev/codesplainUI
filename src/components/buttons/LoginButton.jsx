import React, { PropTypes } from 'react';
import { FlatButton } from 'material-ui';

import GitHubIcon from '../../../res/github-icon.png';

const styles = {
  button: {
    marginTop: '6px',
  },
  img: {
    marginRight: '7px',
    marginBottom: '4px',
  },
  span: {
    color: 'white',
    display: 'inline-flex',
    alignItems: 'center',
  },
};

const LoginButtonLabel = (
  <span style={styles.span}>
    <img
      alt=""
      src={GitHubIcon}
      style={styles.img}
      width="25px"
    />
    Log In
  </span>
);

/*
<LoginButton /> renders as white clickable "LOG IN" text, with a white octocat
icon to its left.
*/

const LoginButton = ({ onClick }) => (
  <FlatButton
    label={LoginButtonLabel}
    onClick={onClick}
    style={styles.button}
  />
);

LoginButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default LoginButton;
