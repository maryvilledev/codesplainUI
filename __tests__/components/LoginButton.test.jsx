import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import LoginButton from '../../src/components/buttons/LoginButton';

const defaultProps = {
  onClick: jest.fn(),
};

describe('<LoginButton />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  it('matches snapshot', () => {
    const wrapper = shallowWithContext(<LoginButton {...defaultProps} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
