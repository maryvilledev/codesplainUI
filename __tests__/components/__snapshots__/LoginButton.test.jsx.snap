exports[`<LoginButton /> matches snapshot 1`] = `
<FlatButton
  disabled={false}
  fullWidth={false}
  label={
    <span
      style={
        Object {
          "alignItems": "center",
          "color": "white",
          "display": "inline-flex",
        }
      }>
      <img
        alt=""
        src="github-icon.png"
        style={
          Object {
            "marginBottom": "4px",
            "marginRight": "7px",
          }
        }
        width="25px" />
      Log In
    </span>
  }
  labelPosition="after"
  labelStyle={Object {}}
  onClick={[Function]}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
  primary={false}
  secondary={false}
  style={
    Object {
      "marginTop": "6px",
    }
  } />
`;
